src/duct/database/sql/hikaricp.clj

```diff
@@ -15,35 +15,38 @@
 (defn- query-parameter-lists [^QueryInfo query-info]
   (mapv query-parameters (.getParametersList query-info)))
 
-(defn- logged-query [^QueryInfo query-info]
+(defn- logged-query [format-query ^QueryInfo query-info]
   (let [query  (.getQuery query-info)
         params (query-parameter-lists query-info)]
-    (into [query] (if (= (count params) 1) (first params) params))))
+    (cond-> (into [query] (if (= (count params) 1) (first params) params))
+      format-query (format-query))))
 
-(defn- logging-listener [logger]
+(defn- logging-listener [logger query-formatter]
   (reify QueryExecutionListener
     (beforeQuery [_ _ _])
     (afterQuery [_ exec-info query-infos]
       (let [elapsed (.getElapsedTime exec-info)
-            queries (mapv logged-query query-infos)]
+            queries (mapv (partial logged-query query-formatter) query-infos)]
```

Comment:
Rather than fold the query-formatter into `logged-query`, instead use: `(comp query-formatter logged-query)`. Then ensure that the default for `query-formatter` is `clojure.core/identity`.